tests/test_gp_interp.py

```diff
@@ -55,1240 +63,258 @@ def get_correlation_length_matrix(correlation_length, g1, g2):
-                    [np.sqrt(var1*var2)*corr, var2]])
-    invLam = np.linalg.inv(cov)
-
-    kernel = "0.1*AnisotropicRBF(invLam={0!r})".format(invLam)
-    #kernel_vk= "0.1*AnisotropicVonKarman(invLam={0!r})".format(invLam)
-
-    print(kernel)
-
-    for npca in npcas:
-        for optimize in optimizes:
-            check_gp(training_data, validation_data, visualization_data, kernel+ "+ WhiteKernel(1e-5, (1e-7, 1e-2))",
-                     npca=npca, optimize=optimize, file_name="test_anisotropic_rbf.fits",
-                     rng=rng, check_config=check_config)
-            check_gp_2pcf(training_data, validation_data, visualization_data, kernel,
-                          npca=npca, anisotropic=True, optimize=optimize, rng=rng, check_config=True)
-
-@timer
-def test_vonkarman_kernel():
-    from scipy import special
-                    
-    corr_lenght = [1.,10.,100.,1000.]
-    kernel_amp = [1e-4,1e-3,1e-2,1.]
-    dist = np.linspace(0,10,100)
-    coord = np.array([dist,dist]).T
-
-    dist = np.linspace(0.01,10,100)
-    coord_corr = np.array([dist,np.zeros_like(dist)]).T
-    
-    def _vonkarman_kernel(param,x):
-        A = (x[:,0]-x[:,0][:,None])
-        B = (x[:,1]-x[:,1][:,None])
-        distance = np.sqrt(A*A + B*B)
-        Filter = distance != 0.
-        K = np.zeros_like(distance)
-        K[Filter] = param[0]**2 * ((distance[Filter]/param[1])**(5./6.) *
-                                   special.kv(-5./6.,2*np.pi*distance[Filter]/param[1]))
-        dist = np.linspace(1e-4,1.,100)
-        div = 5./6.
-        lim0 = special.gamma(div) /(2 * (np.pi**div) )
-        K[~Filter] = param[0]**2 * lim0
-        K /= lim0
-        return K
+    interp = piff.GPInterp(kernel=kernel, optimize=optimize, optimizer=optimizer,
+                           normalize=True, white_noise=0.,
+                           anisotropic=anisotropic, p0=p0,
+                           n_neighbors=4, average_fits=None,
+                           nbins=nbins, min_sep=min_sep, max_sep=max_sep,
+                           logger=None)
+
+    interp.initialize(stars_training)
+    interp.solve(stars=stars_training, logger=None)
+    stars_test = interp.interpolateList(stars_validation)
+
+    xtest = np.array([interp.getProperties(star) for star in stars_validation])
+    y_validation = np.array([star.fit.params for star in stars_validation])
+    y_err = np.sqrt(np.array([star.fit.params_var for star in stars_validation]))
+            
+    y_test = np.array([star.fit.params for star in stars_test])
+
+    np.testing.assert_allclose(y_test, y_validation, atol = 4e-2)
+
+    if optimize:
+        truth_hyperparameters = np.exp(interp._init_theta)
+        fitted_hyperparameters = np.exp(np.array([gp._optimizer._kernel.theta for gp in interp.gps]))
+        np.testing.assert_allclose(fitted_hyperparameters, truth_hyperparameters, rtol = 2.)
+            
+    if plotting:
+        import pylab as plt
```

Comment:
I don't completely understand why, but apparently using pylab is discouraged.  So probably should switch this to
```
import matplotlib.pyplot as plt
```
which I think is equivalent.